CMakeLists.txt

```diff
@@ -80,9 +80,9 @@ if(APPLE AND CMAKE_CXX_COMPILER_ID MATCHES ".*Clang")
     # -Wno-error=unused-command-line-argument is required due to https://llvm.org/bugs/show_bug.cgi?id=7798
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-error=unused-command-line-argument")
 endif()
-set(CMAKE_CXX_FLAGS_RELEASE "-Os -DNDEBUG")
-set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O3 -DNDEBUG")
-set(CMAKE_CXX_FLAGS_SANITIZE "-O1 -g -fno-omit-frame-pointer -fno-optimize-sibling-calls")
```

Comment:
@kkaefer I would like your :eyes: here please. This is the reason why the Ninja build for Windows was initially failing, because Ninja sets some important default flags. Not sure if this fix is ideal because `CMAKE_CXX_FLAGS_RELEASE` will probably set `-O2` on most of the platforms and will do a `-Os` right after. But we should be using `MINSIZEREL` if we want a build optimized for size instead of overriding `RELEASE`.

Finally, we don't override `CMAKE_CC_FLAGS_RELEASE`, so we are mixing optimizations.

Reply:
Ah good catch!